ext/LuxMPIExt.jl

```diff
@@ -0,0 +1,170 @@
+module LuxMPIExt
+
+import Lux: MPIBackend, NCCLBackend, DistributedUtils, __is_extension_loaded, __set_device!,
+            __unwrap_val
+import LuxDeviceUtils: AbstractLuxDevice, LuxCUDADevice, LuxAMDGPUDevice, MPI_CUDA_AWARE,
+                       MPI_ROCM_AWARE, cpu_device
```

Comment:
looks like `LuxDeviceUtils` doesn't have the symbols `MPI_CUDA_AWARE`, `MPI_ROCM_AWARE`:

```julia
(gpu) pkg> st                                                                                                                                                                                                                                                
Status `~/.julia/dev/gpu/Project.toml`                                                                                                                                                                                                                       
  [052768ef] CUDA v5.2.0                                                                                                                                                                                                                                     
  [b2108857] Lux v0.5.22 `~/.julia/dev/Lux`                                                                                                                                                                                                                  
  [d0bbae9a] LuxCUDA v0.3.2                                                                                                                                                                                                                                  
  [34f89e08] LuxDeviceUtils v0.1.17                                                                                                                                                                                                                          
  [da04e1cc] MPI v0.20.19                                                                                                                                                                                                                                    
  [3da0fdf6] MPIPreferences v0.1.10                                                                                                                                                                                                                          
  [3fe64909] NCCL v0.1.1                                                                                                                                                                                                                                     
  [3bd65402] Optimisers v0.3.2                                                                                                                                                                                                                               
  [e88e6eb3] Zygote v0.6.69                                                                                                                                                                                                                                  
  [9a3f8284] Random                                                                                                           
                                                                                                                              
julia> using Lux, MPI                                                                                                         
Precompiling LuxMPIExt                                                                                                                                                                                                                                       
        Info Given LuxMPIExt was explicitly requested, output will be shown live                                              
WARNING: could not import LuxDeviceUtils.MPI_CUDA_AWARE into LuxMPIExt                                                                                                                                                                                       
WARNING: could not import LuxDeviceUtils.MPI_ROCM_AWARE into LuxMPIExt                                                        
ERROR: LoadError: UndefVarError: `MPI_CUDA_AWARE` not defined                                                                 
Stacktrace:                                                                                                                   
 [1] top-level scope                                                                                                          
   @ ~/.julia/dev/Lux/ext/LuxMPIExt.jl:55                                                                                     
 [2] include                                                                                                                  
   @ ./Base.jl:495 [inlined]                      
 [3] include_package_for_output(pkg::Base.PkgId, input::String, depot_path::Vector{String}, dl_load_path::Vector{String}, load_path::Vector{String}, concrete_deps::Vector{Pair{Base.PkgId, UInt128}}, source::Nothing)
   @ Base ./loading.jl:2222                                                                                                   
 [4] top-level scope                                                                                                          
   @ stdin:3                                                                                                                                                                                                                                                 
in expression starting at /home/vedantpu/.julia/dev/Lux/ext/LuxMPIExt.jl:1                                                    
in expression starting at stdin:3                                                                                             
  ✗ Lux → LuxMPIExt                                                                                                           
  0 dependencies successfully precompiled in 3 seconds. 153 already precompiled.                                                                                                                                                                             
```

Reply:
Those symbols are defined in `Lux.jl`. Adding them to line 4 fixes the precompile error.